lib/hsa/mcwamp_hsa.cpp

```diff
@@ -2682,19 +2680,32 @@ class HSADevice final : public KalmarDevice
 
         const auto isa{get_isa_name_from_triple(std::move(triple))};
 
-        hsa_isa_t co_isa{};
-        status = hsa_isa_from_name(isa.c_str(), &co_isa);
+        struct isa_comp_data {
+            hsa_isa_t isa_type;
+            bool is_compatible;
+        } co_data;
+
+        status = hsa_isa_from_name(isa.c_str(), &co_data.isa_type);
         STATUS_CHECK(status, __LINE__);
 
         // Check if the code object is compatible with ISA of the agent
-        bool isCompatible = false;
-        status = hsa_isa_compatible(co_isa, agentISA, &isCompatible);
+        co_data.is_compatible = false;
+        status =
+            hsa_agent_iterate_isas(
+                agent,
+                [](hsa_isa_t agent_isa, void* data) {
+                    isa_comp_data* co_data = static_cast<isa_comp_data*>(data);
+                    if (agent_isa == co_data->isa_type)
+                        co_data->is_compatible = true;
+                    return HSA_STATUS_SUCCESS;
```

Comment:
I think when currrent isa is incompatible then the callback should return `HSA_STATUS_INFO_BREAK` so it goes to the next isa.  See other hsa related callbacks like `get_memory_pools` for example in hcc RT

Reply:
I've added that status_t as well, thanks!